test/common/devCompileOptions.cmake

```
@@ -113,7 +123,9 @@ endif()
 # Clang, AppleClang, ICPX
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
     if(alpaka_ENABLE_WERROR)
-        list(APPEND alpaka_DEV_COMPILE_OPTIONS "-Werror")
+        list(APPEND alpaka_DEV_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:-Werror>
+                                               $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:-Xcompiler -Werror>
+                                               $<$<COMPILE_LANG_AND_ID:CUDA,AppleClang,Clang,IntelLLVM>:-Werror>)
```

Does only nvcc not support `-Werror` directly? Or are you omitting `-Werror` on purpose for CUDA compilers other than NVIDIA, AppleClang, Clang and IntelLLVM? I don't understand why the last clause is necessary.

Why would this not work?
```suggestion
        list(APPEND alpaka_DEV_COMPILE_OPTIONS $<IF:$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>,-Xcompiler -Werror,-Werror>)
```

---

It was a long day...